### ISSUE TITLE
HoconConfigurationLoader.Builder#setRenderOptions(ConfigRenderOptions) throws NoSuchMethodError

### ISSUE THREAD

**Issue by Zerthick** [SpongePowered/Sponge#1391]
**I am currently running...**

- SpongeForge version: spongeforge-1.11.2-2282-6.1.0-BETA-2370.jar
- SpongeVanilla version: 
- Java version: 
- Operating System:

 - [ ] Linux (Ubuntu, Debian, Fedora, etc)
 - [ ] macOS
 - [x] Windows

**What is the expected result?**
The config node should be rendered with the appropriate render options.
**What is the current result?**
Currently a `NoSuchErrorMethod` is thrown.

**Ex:**
```
HoconConfigurationLoader.builder().setRenderOptions(ConfigRenderOptions.concise()).build();
```

Interestingly, when I run my plugin inside of IntelliJ with the SpongeStart plugin, everything works, but when I build a separate jar and run it as a normal pluign this error is thrown.

**Comment by Deamon5550** [comment]
Is the server you are running your built jar in a different version than what you are building your plugin against?

**Comment by Zerthick** [comment]
Nope, same version.

**Comment by Zerthick** [comment]
If you're curious, this is the plugin in question https://github.com/Zerthick/SerializeMe

**Comment by stephan-gh** [comment]
This is probably a SpongeForge issue: Forge bundles its own (outdated) copy of HOCON so we have to relocate the newer version to a different package. Only the production build of SpongeForge has this problem. We should be able to fix it with Minecraft 1.12 because Forge can now upgrade to the latest HOCON version.

**Comment by Zidane** [comment]
@Minecrell 

Was this fixed in 1.12?

**Comment by stephan-gh** [comment]
@Zidane No, Forge decided to keep Scala and configurate at the old versions for 1.12 (see https://github.com/MinecraftForge/MinecraftForge/pull/4029). I can't think of any other way to fix this so we'll need to wait until Forge either updates or removes configurate.

**Comment by Zidane** [comment]
@Minecrell 

We need to figure out something here as its a very annoying issue..

**Comment by mmis1000** [comment]
Found some workaround to make use `HoconConfigurationLoader.Builder#setRenderOptions` possible with spongeForge.
By just relocate reference of `com.typesafe` to `configurate.typesafe` to match the method signature in spongeForge 

```groovy
// !!!!!!!!!!! This works with SpongeForge but breaks SpongeVanilla !!!!!!!!!!!

plugins {
    id 'org.spongepowered.plugin' version '0.8.1'
    id "com.github.johnrengelman.shadow" version "2.0.4"
}

group = pluginGroup
version = pluginVersion

dependencies {
    compile 'org.spongepowered:spongeapi:7.1.0-SNAPSHOT'
}


sponge.plugin.id = pluginId

shadowJar {
    configurations = [project.configurations.shadow]
    classifier = null
    relocate "com.typesafe", "configurate.typesafe"
}

build.dependsOn(shadowJar)
```
